pkg/jobmanager/job.go

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+package jobmanager
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/facebookincubator/contest/pkg/job"
+	"github.com/facebookincubator/contest/pkg/pluginregistry"
+	"github.com/facebookincubator/contest/pkg/test"
+)
+
+func newJob(registry *pluginregistry.PluginRegistry, jobDescriptor *job.Descriptor, resolver stepsResolver) (*job.Job, error) {
+
+	if jobDescriptor == nil {
+		return nil, errors.New("JobDescriptor cannot be nil")
+	}
+	if err := jobDescriptor.Validate(); err != nil {
+		return nil, fmt.Errorf("could not validate job descriptor: %w", err)
+	}
+	tests := make([]*test.Test, 0, len(jobDescriptor.TestDescriptors))
+
+	stepsDescriptors, err := resolver.GetStepsDescriptors()
+	if err != nil {
+		return nil, fmt.Errorf("could not get steps descriptors: %w", err)
+	}
+	if len(stepsDescriptors) != len(jobDescriptor.TestDescriptors) {
+		return nil, fmt.Errorf("length of steps descriptor must match lenght of test descriptors")
+	}
+
+	for index, td := range jobDescriptor.TestDescriptors {
+		thisTestStepsDescriptors := stepsDescriptors[index]
+
+		if err := td.Validate(); err != nil {
+			return nil, fmt.Errorf("could not validate test descriptor: %w", err)
+		}
+		bundlTargetManager, err := registry.NewTargetManagerBundle(&td)
+		if err != nil {
+			return nil, err
+		}
+		bundleTestFetcher, err := registry.NewTestFetcherBundle(&td)
+		if err != nil {
+			return nil, err
+		}
+
+		bundleTest, bundleCleanup, err := newStepsBundles(thisTestStepsDescriptors, registry)
+		if err != nil {
+			return nil, fmt.Errorf("could not create cleanup step bundles: %v", err)
+		}
+		test := test.Test{
+			Name:                thisTestStepsDescriptors.TestName,
+			TargetManagerBundle: bundlTargetManager,
+			TestFetcherBundle:   bundleTestFetcher,
+			TestStepBundles:     bundleTest,
```

Comment:
Sometimes we call it `StepBundle` and sometimes `StepsBundle`. I believe we should pick one to do not mislead anybody.

Reply:
You are right. As far as I can tell, the the `.*StepsBundle.*` nomenclature is used only in `newStepsBundles`, and that shoul dbe `StepBundles`. We should only have `StepBundle` or `StepBundles` (i.e. []StepBundle).